diff --git a/train.py b/train.py
--- a/train.py
+++ b/train.py
@@ -515,11 +515,11 @@ if __name__ == "__main__":
     args = parser.parse_args(sys.argv[1:])
     if args.configs:
         args = apply_config_file(args, args.configs)
-    setup_seed(args.seed)
     args.save_iterations = sorted(set(args.save_iterations + [args.iterations]))
     print("Optimizing " + args.model_path)
 
     # Initialize system state (RNG)
     safe_state(args.quiet)
+    setup_seed(args.seed)
 
     # Start GUI server, configure and run training
